datahub/dbmaintenance/management/commands/update_investment_project_company.py

```
@@ -0,0 +1,112 @@
+        :param intermediate_company: instance of Company or None
+        :param uk_company: instance of Company or None
+        :param uk_company_decided: Boolean
+        :return: True if investment project needs to be updated
+        """
+        return (investment_project.investor_company != investor_company or
+                investment_project.intermediate_company != intermediate_company or
+                investment_project.uk_company != uk_company or
+                investment_project.uk_company_decided != uk_company_decided)
+
+    def _can_update(self, company_ids):
+        """Check if given list of company_ids exists.
+
+        :param: company_ids: list of company ids
+        :return: True if can update, otherwise False
+        """
+        return (
+            Company.objects.values_list('id').filter(id__in=company_ids).count() ==
+            len(company_ids)
+        )
+
+    def get_uk_company_decided(self, uk_company_decided):
+        """
+        :param uk_company_decided: string containing either '1' or '0'
+        :return: Boolean
+        """
+        translate = {
+            '0': False,
+            '1': True,
+        }
+        return translate[uk_company_decided.strip()]
+
+    def _process_row(self, row, simulate=False, **options):
+        """Process one single row."""
+        investment_project = InvestmentProject.objects.get(pk=row['id'])
+        investor_company_id = self._parse_company_id(row['investor_company_id'])
+        intermediate_company_id = self._parse_company_id(row['intermediate_company_id'])
+        uk_company_id = self._parse_company_id(row['uk_company_id'])
+        uk_company_decided = self.get_uk_company_decided(row['uk_company_decided'])
+
+        company_ids = [
+            company_id for company_id in
+            (investor_company_id, intermediate_company_id, uk_company_id,)
+            if company_id is not None
+        ]
+        if not self._can_update(company_ids):
```

you might not need this check, no? They are all foreign keys so if you set a value that doesn't exist, the update command should fail if I'm not missing anything, no?

---

That is true, we don't need that as these are foreign keys. I have removed it.